src/notifications.js

```diff
@@ -12,7 +12,7 @@ class Notifications extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const {notifications} = nextProps;
+    const notifications = nextProps.notifications.toJS();
```

Comment:
Might it work if you used `map()`?

Both native arrays and Immutable List have the map method (plus [decent support](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/map) for native array.map, IE9+). And you're not directly passing the List or array into the _react-notification-system_ component so it shouldn't affect it.

Bonus would also be not having to perform the `toJS()` call on the notifications